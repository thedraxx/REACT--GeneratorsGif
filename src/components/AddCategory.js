import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setinputValue] = useState("Search a GIF!");

    const handeInputChange = (e) =>{
        setinputValue(e.target.value)

        console.log('handeInputChange llamado')
    }

    const handleSubmit = (e) => {
        
        e.preventDefault();
  
        if( inputValue.trim().length > 2){
            setCategories(categories => [ inputValue,...categories,]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit = {handleSubmit}>
            <p>{inputValue}</p>
                <input
                    type="text" 
                    value={inputValue}
                    onChange={ handeInputChange}
                />

        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };
