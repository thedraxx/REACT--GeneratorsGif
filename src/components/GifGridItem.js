import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({tittle,url}) => {

    return (
        <div className = "card animate__animated animate__fadeIn">
           <img src = {url} alt={tittle} />
        </div>
    )
}

GifGridItem.propTypes = {
    tittle: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}



