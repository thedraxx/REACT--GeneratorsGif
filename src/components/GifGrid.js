import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import '../index.css';

export const  GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <div className='Contenedor-titulo'>
          
       
            {loading && <p className= "card animate__animated animate__flash">Loading</p> }
       
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem 
                                key={img.id}
                                {...img}
                            />
                    ))
                }
            </div>
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
