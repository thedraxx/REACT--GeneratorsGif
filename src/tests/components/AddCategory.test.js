import { shallow } from "enzyme"
import '@testing-library/jest-dom';
import { AddCategory } from "../../components/AddCategory";

describe('Pruebas en <AddCategory />', () => {
    
        const setCategories = jest.fn();
        let wrapper = shallow(<AddCategory setCategories = { setCategories } />);

        beforeEach( () => {
            jest.clearAllMocks();
            wrapper = shallow(<AddCategory setCategories = { setCategories } />);
        });

    test('debe de mostrarse correctamete', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: {value}}); 
        expect(wrapper.find('p').text().trim()).toBe(value)
    })

    test('No debe de postear la informacion con onSubmit', () => {
        
        wrapper.find('form').simulate('submit', {preventDefault(){}} );

        expect(setCategories).toHaveBeenCalled();
    })


    test('Debe de llamar el setCategory y limpiar la caja de texto', () => {

        const value = "Hola mundo";

        //Simular el InputChange
        wrapper.find('input').simulate('change', {target: {value}});
 
        //Simular el submit
        wrapper.find('form').simulate('submit', {preventDefault(){}} );

        //SetCategories se debe de haber llamado
        expect(setCategories).toHaveBeenCalled();

        //El valor del input debe de estar ''
        expect(wrapper.find('input').prop('value')).toBe('')
    })
    
    

    
})
