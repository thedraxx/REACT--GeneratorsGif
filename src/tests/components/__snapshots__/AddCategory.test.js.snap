// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrarse correctamete 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Search a GIF!
  </p>
  <input
    onChange={[Function]}
    type="text"
    value="Search a GIF!"
  />
</form>
`;
